### PR TITLE
[gce provider] Add more wrappers for security policy

### PR DESCRIPTION
**Pull request by MrHohn**
**What this PR does / why we need it**:
Adding more wrappers for security policy, mostly for implementing the e2e test.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #NONE 

**Special notes for your reviewer**:
/assign @nicksardo 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/providers/gce/gce_securitypolicy.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+// PatchRuleForBetaSecurityPolicy patches the given security policy
+// rule to a security policy.
+func (gce *GCECloud) PatchRuleForBetaSecurityPolicy(name string, spr *computebeta.SecurityPolicyRule) error {
+	mc := newSecurityPolicyMetricContextWithVersion("add_rule", computeBetaVersion)
+	return mc.Observe(gce.c.BetaSecurityPolicies().PatchRule(context.Background(), meta.GlobalKey(name), spr))
+}
+
+// RemoveRuleFromBetaSecurityPolicy removes rule from a security policy.
+func (gce *GCECloud) RemoveRuleFromBetaSecurityPolicy(name string) error {
+	mc := newSecurityPolicyMetricContextWithVersion("add_rule", computeBetaVersion)
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/gce_securitypolicy.go]: Need to update metric desc.

File: pkg/cloudprovider/providers/gce/gce_securitypolicy.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+// a security policy.
+func (gce *GCECloud) AddRuletoBetaSecurityPolicy(name string, spr *computebeta.SecurityPolicyRule) error {
+	mc := newSecurityPolicyMetricContextWithVersion("add_rule", computeBetaVersion)
+	return mc.Observe(gce.c.BetaSecurityPolicies().AddRule(context.Background(), meta.GlobalKey(name), spr))
+}
+
+// PatchRuleForBetaSecurityPolicy patches the given security policy
+// rule to a security policy.
+func (gce *GCECloud) PatchRuleForBetaSecurityPolicy(name string, spr *computebeta.SecurityPolicyRule) error {
+	mc := newSecurityPolicyMetricContextWithVersion("add_rule", computeBetaVersion)
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/gce_securitypolicy.go]: Need to update metric desc.

File: pkg/cloudprovider/providers/gce/gce_securitypolicy.go (hunk 1 above)
Comment (MrHohn) [reply pkg/cloudprovider/providers/gce/gce_securitypolicy.go]: Sorry..done.

File: pkg/cloudprovider/providers/gce/gce_securitypolicy.go (hunk 2 above)
Comment (MrHohn) [reply pkg/cloudprovider/providers/gce/gce_securitypolicy.go]: Done.

**Pull request merged**